File: src/main/java/sample/simple/WelcomeController.java
package sample.simple;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Collections;
import java.util.Map;

@RestController
public class WelcomeController {

  @GetMapping("/")
  public Map<String, String> welcome() {
    return Collections.singletonMap("message", "Hello, World!");
  }

}

File: src/main/kotlin/sample/simple/WelcomeController.kt
package sample.simple

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import java.util.*

@RestController
open class WelcomeController {

  @GetMapping("/")
  fun welcome(): Map<String, String> {
    return Collections.singletonMap("message", "Hello, World!")
  }

}